File: implementacoes/c/myBib.c
#include <stdio.h>
#include "myBib.h"

int myRand()
{
    // seed é 2048
    static int nSeed = 2048;

    // Usa a seed para gerar um novo valor,
    // É uma constante muito grande e com overflow, deixando bem dificil de prever o proximo numero
    nSeed = (8253729 * nSeed + 2396403); 

    return nSeed  % 32767;
}

File: implementacoes/pascal/myBib.pas
{Biblioteca para padronizar a geracao de numeros aleatorios
nas 4 linguagens}

Unit myBib;
Interface

  Uses sysutils;

  function myRand() : Cardinal;

Implementation
  var
    nSeed : Cardinal;

  function myRand() : Cardinal;
  begin
    nSeed := 2048; {seed vale 2048}

    {Usa a seed para gerar um novo valor. Como e uma constante muito grande e com overflow, fica dificil prever o proximo numero}
    nSeed := (8253729 * nSeed + 2396403); {Nao aceita}

    myRand := (nSeed mod 32767);
  end;

begin

end.
